Sources/SwiftKuery/Connection.swift

```diff
@@ -124,59 +197,109 @@ public protocol Connection {
     func rollback(onCompletion: @escaping ((QueryResult) -> ()))
 
+    /// Rollback the current transaction.
+    ///
+    /// - Returns: A QueryResult representing the result of the operation
+    func rollbackSync() -> QueryResult
+
     /// Create a savepoint.
     ///
     /// - Parameter savepoint: The name to  be given to the created savepoint.
     /// - Parameter onCompletion: The function to be called when the execution of create savepoint command has completed.
     func create(savepoint: String, onCompletion: @escaping ((QueryResult) -> ()))
 
+    /// Create a savepoint.
+    ///
+    /// - Parameter savepoint: The name to  be given to the created savepoint.
+    /// - Returns: A QueryResult representing the result of the operation
+    func createSync(savepoint: String) -> QueryResult
+
     /// Rollback the current transaction to the specified savepoint.
     ///
     /// - Parameter to savepoint: The name of the savepoint to rollback to.
     /// - Parameter onCompletion: The function to be called when the execution of rolback transaction command has completed.
     func rollback(to savepoint: String, onCompletion: @escaping ((QueryResult) -> ()))
 
+    /// Rollback the current transaction to the specified savepoint.
+    ///
+    /// - Parameter to savepoint: The name of the savepoint to rollback to.
+    /// - Returns: A QueryResult representing the result of the operation
+    func rollbackSync(to savepoint: String) -> QueryResult
+
     /// Release a savepoint.
     ///
     /// - Parameter savepoint: The name of the savepoint to release.
     /// - Parameter onCompletion: The function to be called when the execution of release savepoint command has completed.
-    func release(savepoint: String, onCompletion: @escaping ((QueryResult) -> ()))    
+    func release(savepoint: String, onCompletion: @escaping ((QueryResult) -> ()))
+
+    /// Release a savepoint.
+    ///
+    /// - Parameter savepoint: The name of the savepoint to release.
+    /// - Returns: A QueryResult representing the result of the operation
+    func releaseSync(savepoint: String) -> QueryResult
 }
 
 public extension Connection {
+
     func execute(query: Query, parameters: [String:Any?], onCompletion: @escaping ((QueryResult) -> ())) {
-        do {
-            let databaseQuery = try query.build(queryBuilder: queryBuilder)
-            let (convertedQuery, namedToNumbered, count) = Utils.convertNamedParametersToNumbered(query: databaseQuery, queryBuilder: queryBuilder)
-            var numberedParameters: [Any?] = Array(repeating: nil, count: count)
-            for (parameterName, parameterValue) in parameters {
-                if let numbers = namedToNumbered[parameterName] {
-                    for number in numbers {
-                        numberedParameters[number - 1] = parameterValue
+        DispatchQueue.global().async {
```

Comment:
I'm not sure using the global queue is a great idea - should we create a "Kuery queue"?

Reply:
The issue with this is where we would "anchor" our queue. I'll do some thinking and see if I can come up with a suitable solution.

Reply:
Looking around it seems that work placed on custom queues will make its way onto a global queue anyway so this is probably better left as is.

Reply:
We will leave this as is given the desire to take another look at how we interact with the database